lit_gpt/config.py

```diff
@@ -83,6 +85,17 @@ def from_name(cls, name: str, **kwargs: Any) -> Self:
         conf_dict.update(kwargs)
         return cls(**conf_dict)
 
+    @classmethod
+    def from_json(cls, path: Union[str, Path], **kwargs: Any) -> Self:
+        with open(path, encoding="utf-8") as fp:
+            json_kwargs = json.load(fp)
+        if "condense_ratio" in json_kwargs:  # legacy name
+            json_kwargs["rope_condense_ratio"] = json_kwargs.pop("condense_ratio")
+        if "condense_ratio" in kwargs:  # legacy name
+            kwargs["rope_condense_ratio"] = kwargs.pop("condense_ratio")
+        json_kwargs.update(kwargs)
```

Comment:
This one changes the previous behavior: before the default values were overridden by the values from the config file, and now it's the opposite.
In my books, values from the config file are more important.

Reply:
We have cases for both: finetune/lora overrode the args whereas xla/generate/adapter overrode the config.

I think it's more likely that users add an arg to the source code than modify the config json, in that case I prefer the current behaviour